apps/hpc-ftsadmin/src/app/utils/codecs.ts

```diff
@@ -0,0 +1,65 @@
+import * as t from 'io-ts';
+import { util } from '@unocha/hpc-data';
+import {
+  DEFAULT_FLOW_TABLE_HEADERS,
+  DEFAULT_KEYWORD_TABLE_HEADERS,
+  DEFAULT_ORGANIZATION_TABLE_HEADERS,
+  FlowHeaderID,
+  KeywordHeaderID,
+  OrganizationHeaderID,
+  TableHeadersProps,
+} from './table-headers';
+
+const PARAMS_CODEC = t.intersection([
+  t.type({
+    page: util.INTEGER_FROM_STRING,
+    rowsPerPage: util.INTEGER_FROM_STRING,
+    orderDir: t.keyof({
+      ASC: 'ASC',
+      DESC: 'DESC',
+    }),
+    filters: t.string,
+    tableHeaders: t.string,
+  }),
+  t.partial({
+    prevPageCursor: util.INTEGER_FROM_STRING,
+    nextPageCursor: util.INTEGER_FROM_STRING,
+  }),
```

Comment:
It should be noted that paging doesn't work without cursors, even though it appears by looking in the UI label saying something like "51–100 of 395432". This label is constructed via `page` query param, but API doesn't return proper results without cursors.

This problem is not created by this PR, but already exists on `develop`. We should ticket it separately.